wpcom-helper.php

```diff
@@ -67,29 +85,55 @@ function wpcom_amp_get_stats_extras_url() {
 	return $url;
 }
 
-add_action( 'pre_amp_render', function() {
-	add_filter( 'post_flair_disable', '__return_true', 99 );
-	remove_filter( 'the_title', 'widont' );
+add_action( 'pre_amp_render', 'wpcom_helper_pre_amp_render' );
+
+function wpcom_helper_pre_amp_render() {
+	if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) { // WP.com only.
+		add_filter( 'post_flair_disable', '__return_true', 99 );
+
+		remove_filter( 'the_title', 'widont' );
+
+		remove_filter( 'pre_kses', array( 'Filter_Embedded_HTML_Objects', 'filter' ), 11 );
+		remove_filter( 'pre_kses', array( 'Filter_Embedded_HTML_Objects', 'maybe_create_links' ), 100 );
+	} else { // Jetpack only.
```

Comment:
I think we should have an explicit Jetpack check here rather than rely on the `else`. Also, can we split up the wpcom and Jetpack bits into two separate functions?